scripts/hints.js

```diff
@@ -35,7 +35,92 @@ if (hint) {
 }
 
 // Provide an auxiliary hint.
-if (process.env.SKIP_TS_CHECK) {
+const tsCheckAffects = ['build', 'watch', 'test', 'test-autobuild'];
+if (process.env.SKIP_TS_CHECK && tsCheckAffects.includes(hintName)) {
```

Comment:
Since we're reusing the hints script, this change is just to avoid showing the TS hint during npm install, where it's irrelevant.